feat: display users and devices E2E identity verification badges

przemvs opened the pull request:
## Description
Epic PR for UI for E2EI.

Displaying verification shields.

Possible to display:
- Proteus
- Valid MLS
- Expire soon MLS certificate
- Expired MLS certificate
- Revoked MLS certificate

Now we have only prepared functionality to use new verification shields. Proper displaying will be prepared in this epic: https://wearezeta.atlassian.net/browse/WPB-2253.

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

aweiss-dev committed:
feat: add route for e2eIdentity Landingpage, add server redirect

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

aweiss-dev committed:
feat: add E2EI Feature flag

aweiss-dev committed:
feat: add info

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

aweiss-dev committed:
wip

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

aweiss-dev committed:
Merge branch dev into feat/e2ei_oicd

aweiss-dev committed:
feat: add loading, error and done modals to the flow

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

aweiss-dev committed:
fix: update ignore

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

aweiss-dev committed:
featr: finish modal implementation and setup tests

aweiss-dev committed:
fix: remove yalc

aweiss-dev committed:
chore: rename delayTimer callbacks for clarity

aweiss-dev committed:
Co-authored-by: Amir Ghezelbash <thisisamir98@users.noreply.github.com>

aweiss-dev committed:
feat: change redirect route from /auth/oidc to /auth to circumvent cookie problem

aweiss-dev committed:
chore: merge dev into feat/e2ei_oicd

aweiss-dev committed:
fix: lockfile & types

aweiss-dev committed:
feat: get the certificate is working

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

aweiss-dev committed:
feat: remove unused page

aweiss-dev committed:
Merge branch dev into feat/e2ei_oicd

aweiss-dev committed:
feat: change api of E2EI

przemvs committed:
feat: Conversations shields

aweiss-dev committed:
feat: clear uri from redirect valuesw

przemvs committed:
Prepare shields

przemvs committed:
remove unused files

przemvs committed:
update wire-web-packages and remove icons from wire-webapp

aweiss-dev committed:
Merge branch dev into feat/e2ei_oicd

aweiss-dev committed:
refactor: move parts of the code out of webapp

aweiss-dev committed:
fix: router behaviour for oauth

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

przemvs committed:
feat: add base tests

aweiss-dev committed:
feat: add UI functionality (redirects) to thr webapp

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

przemvs committed:
Merge branch 'dev' into feat/e2ei_oicd

aweiss-dev committed:
chore: rebuild yarn.lock

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

przemvs committed:
Merge branch 'dev' into feat/e2ei_oicd

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

aweiss-dev committed:
feat: add a new featureChangeHandler to start the E2EI flow

aweiss-dev committed:
feat: clear oidc progress at error

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

przemvs committed:
feat: replace url param after get certificate

aweiss-dev committed:
fix: clean up after success or failed try.

aweiss-dev committed:
chore: move util functions

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

przemvs committed:
Merge branch 'dev' into feat/e2ei_oicd

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

aweiss-dev committed:
fix: yarn errors

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

phoenixhdd committed:
feat: Show MLS Info on devices (#15637)

* feature: Show MLS Info on devices

* Prepare mls details in right sidebar and device details

* remove unecessary code

* feat: add mls fingerprint

* fix todos

* feat: add translations

* code review changes

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

atomrc commented on src/script/components/Badges/Badges.tsx:
```diff
@@ -0,0 +1,162 @@
+  marginRight: '4px',
+});
+
+export const Badges: React.FC<BadgesProps> = ({
```
suggestion: I don't think `Badges` properly conveys the functionality here. How about naming it `VerificationBadge`?

przemvs committed:
rename badges to verificationbadges

atomrc commented on src/script/components/panel/UserDetails.tsx:
```diff
@@ -94,12 +96,12 @@ export const UserDetailsComponent: React.FC<UserDetailsProps> = ({
+        {/*    className="panel-participant__head__verified-icon"*/}
+        {/*    data-uie-name="status-verified-participant"*/}
+        {/*  />*/}
+        {/*)}*/}
```
how about this?

przemvs replied:
I'll keep it to know where should be this icon and forgot to remove.

atomrc reviewed (approved):
LGTM

przemvs committed:
cr fixes

PatrykBuniX commented on src/script/page/MainContent/panels/preferences/devices/components/DetailedDevice.tsx:
```diff
@@ -19,52 +19,40 @@
-      <p className="preferences-devices-id">
-        <strong>{t('preferencesDevicesId')}</strong>
+export const DetailedDevice: React.FC<DeviceProps> = ({device, fingerprint, showVerificationStatus = true}) => {
+  const isProteusVerified = true;
```
**question**: This one will be handled is some following PR i assume?

przemvs replied:
Yup, we have epic task for shields :)

phoenixhdd committed:
feat: Show E2E Identity certificate info (#15677)

* feature: Show MLS Info on devices

* Prepare mls details in right sidebar and device details

* remove unecessary code

* feature: Show E2E Identity certificate info

* remove unecessary code

* fix: change design for serial number

* feat: add mls fingerprint

* fix todos

* connect with core and hide serial number

* fix: rename to e2ei from e2e

* feat: See E2EI certificate details [WPB-2213] (#15745)

* feat: See E2EI certificate details

* feat: Add certificate details modal

* fix: create proper styles for modal

* fix: add comments

* fixes: remove unecessary code

* fix: codereview changes

* feat: add translations

* code review changes

* get e2ei certificate

* feat: Download E2EI certificate (#15794)

* feat: Download E2EI certificate

* cr changes

* Add certificate details

* Other devices e2ei status

* cr fixes

* CR Changes

* feat: Add expires soon badge (#15899)

* feat: Add expires soon badge

* Add comment

przemvs committed:
fixes

aweiss-dev committed:
feat: manage conversation verification state for mls [WPB-482] (#15889)

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

aweiss-dev committed:
Merge branch 'feat/e2ei_oicd' of github.com:wireapp/wire-webapp into feat/e2ei_oicd

aweiss-dev committed:
Merge branch 'dev' into feat/e2ei_oicd

przemvs committed:
Merge branch 'feat/e2ei_oicd' into feature/WPB-482-conversation-shields

przemvs committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

przemvs committed:
revert loadable component

przemvs committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

przemvs committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

phoenixhdd committed:
feat: Indicate client with valid certificate (#16018)

* feat: Indicate client with valid certificate

* remove unecessary code

przemvs committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

przemvs committed:
yarn.lock file update

przemvs committed:
yarn.lock file update

przemvs committed:
update yarn.locks

przemvs committed:
lint fixes

przemvs committed:
test fixes

aweiss-dev committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

phoenixhdd committed:
feat: Add MLS Verification Badges (#16029)

* feat: Add MLS Verification Badges

* fix:  remove yalc import

---------

Co-authored-by: Adrian Weiß <gitlab@weissadrian.com>

PatrykBuniX commented on src/script/components/VerificationBadges/VerificationBadges.tsx:
```diff
@@ -0,0 +1,161 @@
+    <div className="conversation-badges" style={{display: 'flex', alignItems: 'center', gap: '6px'}}>
+      {showMLSBadge && (
+        <div style={badgeWrapper}>
+          {displayTitle && <span style={title(true)}>Verified (End-to-end Identity)</span>}
```
Hardcoded string spotted 👀, it should be added to translations file i believe, same thing for https://github.com/wireapp/wire-webapp/pull/15575/files#diff-48eda40ee53d9abb4cc638196bd62cac3b88ec9fbc209e69f359c892a7692469R147

przemvs replied:
Fixed ;)

PatrykBuniX commented on src/script/page/MainContent/panels/preferences/DevicesPreferences/components/DeviceDetailsPreferences/DeviceDetailsPreferences.tsx:
```diff
@@ -139,9 +150,16 @@ const DeviceDetailsPreferences: React.FC<DevicesPreferencesProps> = ({
+          <header className="preferences-header">
+            <hr className="preferences-separator" />
+          </header>
+        </section>
```
What's this one? New section for just a separator? 🤔

przemvs replied:
This now is changed, now we have only separator, don't have whole section :)

PatrykBuniX commented on src/script/util/TypePredicateUtil.ts:
```diff
@@ -43,3 +44,8 @@ export function isConversationEntity(conversation: any): conversation is Convers
+// TODO: Temporary checking for mls_public_keys
+export function hasMlsPublicKeys(client: any): client is RegisteredClient {
+  return !!client && 'mls_public_keys' in client;
+}
```
We should improve this guard.

przemvs committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

przemvs committed:
test fixes

przemvs committed:
Add translations and fix section

przemvs committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

przemvs committed:
fix

przemvs committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

przemvs committed:
restore files from dev

phoenixhdd committed:
feat: Conversation shields improvements (#16163)

* feat: Conversation shields improvements

* CR fixes

przemvs committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

przemvs commented on src/script/components/panel/UserDetails.test.tsx:
```diff
@@ -54,20 +53,6 @@ describe('UserDetails', () => {
     expect(queryByTestId('status-admin')).toBeNull();
   });
 
-  it('shows a verified icon when all clients from the self user are verified and all clients of the other participant are verified', () => {
-    const otherParticipant = new User(createUuid());
-    const verifiedClient = new ClientEntity(false, null);
-    verifiedClient.meta.isVerified?.(true);
-    otherParticipant.devices.push(verifiedClient);
-
-    const props = {isGroupAdmin: true, isSelfVerified: true, participant: otherParticipant};
-
-    const {queryByTestId} = render(<UserDetails {...props} />);
-
-    expect(queryByTestId('status-verified-participant')).not.toBeNull();
-    expect(queryByTestId('status-admin')).not.toBeNull();
-  });
-
```
This part is unnecessary, we have now VerificationBadges component for this.

przemvs commented on src/script/components/userDevices/DeviceCard.test.tsx:
```diff
@@ -37,36 +37,6 @@ function createClientEntity(clientEntity: Partial<ClientEntity>): ClientEntity {
 }
 
 describe('DeviceCard', () => {
-  it('renders desktop icon for desktop clients', async () => {
-    const props = {
-      device: createClientEntity({
-        class: ClientClassification.DESKTOP,
-        meta: {
-          isVerified: ko.observable<boolean>(false),
-        },
-      }),
-      showIcon: true,
-    };
-
-    const {getByTestId} = render(<DeviceCard {...props} />);
-    expect(getByTestId('status-desktop-device')).not.toBeNull();
-  });
```
We don't display desktop icon now.

![image](https://github.com/wireapp/wire-webapp/assets/13432884/8bd6f486-30b6-437b-abb5-637494328e39)

przemvs commented on src/script/components/userDevices/DeviceCard.test.tsx:
```diff
@@ -37,36 +37,6 @@ function createClientEntity(clientEntity: Partial<ClientEntity>): ClientEntity {
-    expect(getByTestId('status-desktop-device')).not.toBeNull();
-  });
-
-  it('renders mobile devices icon for non-desktop clients', async () => {
-    const props = {
-      device: createClientEntity({
-        class: ClientClassification.PHONE,
-        meta: {
-          isVerified: ko.observable<boolean>(false),
-        },
-      }),
-      showIcon: true,
-    };
-
-    const {getByTestId} = render(<DeviceCard {...props} />);
-    expect(getByTestId('status-mobile-device')).not.toBeNull();
-  });
```
We don't display mobile icon now.

![image](https://github.com/wireapp/wire-webapp/assets/13432884/765dc0e0-c534-4e2e-aed2-949292e003f5)

przemvs commented on src/script/components/userDevices/DeviceCard.test.tsx:
```diff
@@ -82,36 +52,4 @@ describe('DeviceCard', () => {
     expect(getByTestId('disclose-icon')).not.toBeNull();
   });
-
-  it('shows verified icon', async () => {
-    const props = {
-      device: createClientEntity({
-        class: ClientClassification.PHONE,
-        meta: {
-          isVerified: ko.observable<boolean>(true),
-        },
-      }),
-      showIcon: true,
-      showVerified: true,
-    };
-
-    const {getByTestId} = render(<DeviceCard {...props} />);
-    expect(getByTestId('user-device-verified')).not.toBeNull();
-  });
-
-  it('shows unverified icon', async () => {
-    const props = {
-      device: createClientEntity({
-        class: ClientClassification.PHONE,
-        meta: {
-          isVerified: ko.observable<boolean>(false),
-        },
-      }),
-      showIcon: true,
-      showVerified: true,
-    };
-
-    const {getByTestId} = render(<DeviceCard {...props} />);
-    expect(getByTestId('user-device-not-verified')).not.toBeNull();
-  });
```
Verified Icon displaying now in VerificationBadges.

przemvs commented on src/script/util/certificateDetails.ts:
```diff
@@ -0,0 +1,74 @@
+import {MLSStatuses} from 'Components/VerificationBadges';
+
+// TODO: Will be changed when I get information how much hours before we need to display this status.
+const EXPIRATION_HOURS = 4;
```
We still waiting for information.

przemvs committed:
revert unnecessary changes

phoenixhdd committed:
feat: Prepare core e2ei service method (#16164)

* feat: Prepare core e2ei service method

* fix: core mock in e2eiHandler tests

* remove duplicated code

* fix tests

---------

Co-authored-by: Adrian Weiß <gitlab@weissadrian.com>

przemvs committed:
Merge branch 'dev' into feature/WPB-482-conversation-shields

Commit:
Merge branch 'dev' into feature/WPB-482-conversation-shields

atomrc committed:
fix: Improve user verification badges computation (#16218)

Commit:
fix spelling of enroll

atomrc committed:
refactor: Use corecrypto to get devices verification state (#16220)

Commit:
Merge branch 'dev' into feature/WPB-482-conversation-shields

atomrc committed:
feat: rely on CoreCrypto to get device full identity (#16221)

atomrc committed:
fix: Do not show current device proteus verification state (#16222)

dependabot[bot] committed:
chore(deps): Bump crowdin/github-action from 1.15.1 to 1.15.2 (#16227)

Bumps [crowdin/github-action](https://github.com/crowdin/github-action) from 1.15.1 to 1.15.2.
- [Release notes](https://github.com/crowdin/github-action/releases)
- [Commits](https://github.com/crowdin/github-action/compare/v1.15.1...v1.15.2)

---
updated-dependencies:
- dependency-name: crowdin/github-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump @wireapp/commons from 5.2.2 to 5.2.3 in /server (#16229)

Bumps [@wireapp/commons](https://github.com/wireapp/wire-web-packages) from 5.2.2 to 5.2.3.
- [Commits](https://github.com/wireapp/wire-web-packages/compare/@wireapp/commons@5.2.2...@wireapp/commons@5.2.3)

---
updated-dependencies:
- dependency-name: "@wireapp/commons"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump the eslint group with 2 updates (#16230)

Bumps the eslint group with 2 updates: [eslint](https://github.com/eslint/eslint) and [@types/eslint](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/eslint).


Updates `eslint` from 8.53.0 to 8.54.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.53.0...v8.54.0)

Updates `@types/eslint` from 8.44.4 to 8.44.7
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/eslint)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: eslint
- dependency-name: "@types/eslint"
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: eslint
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump the datadog group with 2 updates (#16231)

Bumps the datadog group with 2 updates: [@datadog/browser-logs](https://github.com/DataDog/browser-sdk/tree/HEAD/packages/logs) and [@datadog/browser-rum](https://github.com/DataDog/browser-sdk/tree/HEAD/packages/rum).


Updates `@datadog/browser-logs` from 5.2.0 to 5.3.0
- [Changelog](https://github.com/DataDog/browser-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/DataDog/browser-sdk/commits/v5.3.0/packages/logs)

Updates `@datadog/browser-rum` from 5.2.0 to 5.3.0
- [Changelog](https://github.com/DataDog/browser-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/DataDog/browser-sdk/commits/v5.3.0/packages/rum)

---
updated-dependencies:
- dependency-name: "@datadog/browser-logs"
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: datadog
- dependency-name: "@datadog/browser-rum"
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: datadog
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @wireapp/copy-config from 2.1.10 to 2.1.11 (#16232)

Bumps [@wireapp/copy-config](https://github.com/wireapp/wire-web-packages) from 2.1.10 to 2.1.11.
- [Commits](https://github.com/wireapp/wire-web-packages/compare/@wireapp/copy-config@2.1.10...@wireapp/copy-config@2.1.11)

---
updated-dependencies:
- dependency-name: "@wireapp/copy-config"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump react-intl from 6.5.4 to 6.5.5 (#16233)

Bumps [react-intl](https://github.com/formatjs/formatjs) from 6.5.4 to 6.5.5.
- [Release notes](https://github.com/formatjs/formatjs/releases)
- [Commits](https://github.com/formatjs/formatjs/compare/react-intl@6.5.4...react-intl@6.5.5)

---
updated-dependencies:
- dependency-name: react-intl
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump @lexical/react from 0.12.2 to 0.12.4 (#16235)

Bumps [@lexical/react](https://github.com/facebook/lexical/tree/HEAD/packages/lexical-react) from 0.12.2 to 0.12.4.
- [Release notes](https://github.com/facebook/lexical/releases)
- [Changelog](https://github.com/facebook/lexical/blob/main/CHANGELOG.md)
- [Commits](https://github.com/facebook/lexical/commits/v0.12.4/packages/lexical-react)

---
updated-dependencies:
- dependency-name: "@lexical/react"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump @wireapp/core from 42.19.2 to 42.20.0 (#16234)

Bumps [@wireapp/core](https://github.com/wireapp/wire-web-packages) from 42.19.2 to 42.20.0.
- [Commits](https://github.com/wireapp/wire-web-packages/compare/@wireapp/core@42.19.2...@wireapp/core@42.20.0)

---
updated-dependencies:
- dependency-name: "@wireapp/core"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @types/node from 20.9.0 to 20.9.2 (#16239)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.9.0 to 20.9.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump react-router-dom from 6.18.0 to 6.19.0 (#16242)

Bumps [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom) from 6.18.0 to 6.19.0.
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router-dom/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router-dom@6.19.0/packages/react-router-dom)

---
updated-dependencies:
- dependency-name: react-router-dom
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @formatjs/cli from 6.2.3 to 6.2.4 (#16243)

Bumps [@formatjs/cli](https://github.com/formatjs/formatjs) from 6.2.3 to 6.2.4.
- [Release notes](https://github.com/formatjs/formatjs/releases)
- [Commits](https://github.com/formatjs/formatjs/compare/@formatjs/cli@6.2.3...@formatjs/cli@6.2.4)

---
updated-dependencies:
- dependency-name: "@formatjs/cli"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @testing-library/react from 14.1.0 to 14.1.2 (#16240)

Bumps [@testing-library/react](https://github.com/testing-library/react-testing-library) from 14.1.0 to 14.1.2.
- [Release notes](https://github.com/testing-library/react-testing-library/releases)
- [Changelog](https://github.com/testing-library/react-testing-library/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testing-library/react-testing-library/compare/v14.1.0...v14.1.2)

---
updated-dependencies:
- dependency-name: "@testing-library/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump @wireapp/commons from 5.2.2 to 5.2.3 (#16241)

Bumps [@wireapp/commons](https://github.com/wireapp/wire-web-packages) from 5.2.2 to 5.2.3.
- [Commits](https://github.com/wireapp/wire-web-packages/compare/@wireapp/commons@5.2.2...@wireapp/commons@5.2.3)

---
updated-dependencies:
- dependency-name: "@wireapp/commons"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump @wireapp/react-ui-kit from 9.9.12 to 9.10.0 (#16245)

Bumps [@wireapp/react-ui-kit](https://github.com/wireapp/wire-web-packages) from 9.9.12 to 9.10.0.
- [Commits](https://github.com/wireapp/wire-web-packages/compare/@wireapp/react-ui-kit@9.9.12...@wireapp/react-ui-kit@9.10.0)

---
updated-dependencies:
- dependency-name: "@wireapp/react-ui-kit"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump react-router from 6.18.0 to 6.19.0 (#16236)

Bumps [react-router](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router) from 6.18.0 to 6.19.0.
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router@6.19.0/packages/react-router)

---
updated-dependencies:
- dependency-name: react-router
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @faker-js/faker from 8.2.0 to 8.3.1 (#16246)

Bumps [@faker-js/faker](https://github.com/faker-js/faker) from 8.2.0 to 8.3.1.
- [Release notes](https://github.com/faker-js/faker/releases)
- [Changelog](https://github.com/faker-js/faker/blob/next/CHANGELOG.md)
- [Commits](https://github.com/faker-js/faker/compare/v8.2.0...v8.3.1)

---
updated-dependencies:
- dependency-name: "@faker-js/faker"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump @lexical/history from 0.12.2 to 0.12.4 (#16244)

Bumps [@lexical/history](https://github.com/facebook/lexical/tree/HEAD/packages/lexical-history) from 0.12.2 to 0.12.4.
- [Release notes](https://github.com/facebook/lexical/releases)
- [Changelog](https://github.com/facebook/lexical/blob/main/CHANGELOG.md)
- [Commits](https://github.com/facebook/lexical/commits/v0.12.4/packages/lexical-history)

---
updated-dependencies:
- dependency-name: "@lexical/history"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump lexical from 0.12.2 to 0.12.4 (#16238)

Bumps [lexical](https://github.com/facebook/lexical/tree/HEAD/packages/lexical) from 0.12.2 to 0.12.4.
- [Release notes](https://github.com/facebook/lexical/releases)
- [Changelog](https://github.com/facebook/lexical/blob/main/CHANGELOG.md)
- [Commits](https://github.com/facebook/lexical/commits/v0.12.4/packages/lexical)

---
updated-dependencies:
- dependency-name: lexical
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

otto-the-bot committed:
chore: Update translations (#16228)

atomrc committed:
chore: Setup user verification badges (#16226)

PatrykBuniX committed:
fix: do not hide conversation when its content is cleared (#16247)

thisisamir98 committed:
fix: show confirm modal when calling a conversation from wrapper top bar menu (WPB-5445) (#16248)

* fix: show confirm modal when calling a conversation from wrapper top bar menu (WPB-5445)

* change number

* bring all implemenation to calling view model

* use MAX_USERS_TO_PING_WITHOUT_ALERT env

* change count to 4

phoenixhdd committed:
feat: Show E2EI Verification system message (#16224)

* feat: Show E2EI Verification system message

* Add second line message

* render e2ei verification system message

* remove unused code

* show system message

* fix for display verification message

* fix for displaying cell state description

* suggestion fix

otto-the-bot committed:
chore: Update translations (#16249)

atomrc committed:
chore: cleanup applock extra observables (#16250)

PatrykBuniX committed:
runfix: cleared conversation visible (#16252)

* runfix: do not filter removed and cleared conversations

* runfix: update cleared timestamp after clearing a conversation

* runfix: update cleared timestamp on member leave event

* runfix: clear content timestamp

* runfix: do not delete left and cleared conversations from the store

* test: update tests

* runfix: do not change active conversation if update happend in cleared convo

aweiss-dev committed:
feat: use backend provided idp (#16254)

* feat: use backend provided idp

* feat: Update Core

atomrc committed:
runfix: Correctly show user/devices verification state (#16253)

Commit:
upgrade core

Commit:
remove unused x509 certificate lib

aweiss-dev committed:
feat(E2EI): add possibility to refresh a active certificate (#16288)

Commit:
Merge branch 'dev' into feature/WPB-482-conversation-shields

Commit:
Merge branch 'feature/WPB-482-conversation-shields' of github.com:wireapp/wire-webapp into feature/WPB-482-conversation-shields

Commit:
hide e2ei badges when not in an e2ei env

Commit:
fix: fix dark mode

Commit:
fix: fix mls envs

Commit:
fix types

Commit:
refactor: use hook for identity computation

Commit:
style improvements

Commit:
revert gitignore changes

Commit:
Merge branch 'dev' into feature/WPB-482-conversation-shields
